src/System.Management.Automation/engine/hostifaces/LocalConnection.cs

```diff
@@ -867,24 +838,25 @@ private void DoCloseHelper()
                 }
 
                 if ((hostRunspace == null) || (this == hostRunspace))
-                {                   
+                {
+                    bool openRunspaces = false;
                     // We should close transcripting only if we are closing the last opened runspace.
                     foreach (Runspace runspace in RunspaceList)
                     {
                         // At this stage, the last opened runspace should be at closing state.
                         if (runspace.RunspaceStateInfo.State == RunspaceState.Opened)
                         {
-                            return;
+                            openRunspaces = true;
+                            break;
                         }
                     }
 
                     PSHostUserInterface host = executionContext.EngineHostInterface.UI;
-                    if (host != null)
+                    if (host != null && openRunspaces == false)
                     {
                         host.StopAllTranscribing();
+                        AmsiUtils.Uninitialize();
```

Comment:
`AmsiUtils.Uninitialize()` should be called even if `host == null`.
It looks to me we don't need to change this method. Here is the logic of the existing method:
If there is still an open Runsapce, then this method will just return; if there is no open Runspace, then the method will continue: call `StopAllTranscribing` if `host != null`, and then call `AmsiUtils.Uninitialize()`.

Reply:
will revert previous change